src/physics/euler.cpp

```diff
@@ -152,20 +153,18 @@ ::convective_source_term (
 
 template <int dim, int nstate, typename real>
 template<typename real2>
-void Euler<dim,nstate,real>::check_positive_density(real2 &density) const {
-    if (density < 0.0) {
-        this->pcout << "WARNING: Negative density encountered! Setting density as BIG_NUMBER." << std::endl;
-        density = BIG_NUMBER;
+bool Euler<dim,nstate,real>::check_positive_quantity(real2 &qty, const std::string qty_name) const {
+    bool qty_is_positive;
+    if (qty < 0.0) {
+        // Refer to base class for non-physical results handling
+        // Compiler complains unless we explicitly specify "template"
```

Comment:
```suggestion
```

Reply:
this is standard when you have templated member functions in a parent class; you'll see it all throughout NavierStokes